src/screens/TopicPageScreen/elements/ButtonFollowing.js

```diff
@@ -1,36 +1,42 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import {View, TouchableNativeFeedback, Text, StyleSheet} from 'react-native';
 import {fonts, normalize, normalizeFontSize} from '../../../utils/fonts';
 import {COLORS} from '../../../utils/theme';
 
-const ButtonFollowing = ({handleSetUnFollow}) => {
+const ButtonFollowing = ({handleSetUnFollow, followType = 'signed'}) => {
   return (
     <TouchableNativeFeedback onPress={handleSetUnFollow}>
-      <View style={styles.buttonFollowing}>
-        <Text style={styles.textButtonFollowing}>Joined</Text>
+      <View style={styles.buttonFollowing(followType)}>
+        <Text style={styles.textButtonFollowing(followType)}>Joined</Text>
       </View>
     </TouchableNativeFeedback>
   );
 };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `ButtonFollowing` component now accepts a new prop `followType`. It's good that you have provided a default value for this prop. However, it would be better to handle the case when `handleSetUnFollow` is not provided as a prop. This will prevent potential crashes if the function is not passed in.

```diff
const ButtonFollowing = ({handleSetUnFollow, followType = 'signed'}) => {
   return (
     <TouchableNativeFeedback onPress={handleSetUnFollow ? handleSetUnFollow : () => {}}>
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->